### DISCUSSION TITLE
Would it be possible to use `typst watch` without keeping all images in RAM?

### DISCUSSION THREAD

**Discussion by ghost** [typst/typst#5279] [General]
When including big images like [here](https://github.com/typst/typst/issues/5278), I noticed that typst will use a high amount of memory (1 gb per very big image of 60 mb filesize) and it stays at this point all the time. Even when the image is never changed again after it was included once, it stays in memory.
At the same time, I'm using SumatraPDF live view and this will dynamically allocate and deallocate memory w.r.t. the current view region, so SumatraPDF will not stay at a high RAM usage.
Could typst do this similiarly or would it be too difficult w.r.t. the architecture?

**Comment by laurmaedje** [comment]
Typst could be optimized to do that, yes.
